client/login/wp-login/index.jsx

```diff
@@ -79,25 +85,31 @@ class Login extends React.Component {
 		}
 	}
 
-	footerContent() {
+	footerLinks() {
 		const {
 			magicLoginEnabled,
 			magicLoginView,
 			translate,
 		} = this.props;
-		let loginLink;
 
-		if ( magicLoginEnabled && ! magicLoginView ) {
-			loginLink = <a href="#" onClick={ this.props.onMagicLoginRequestClick }>{ translate( 'Email me a login link' ) }</a>;
+		const footerLinks = [];
+
+		if ( magicLoginEnabled ) {
+			if ( magicLoginView === REQUEST_FORM ) {
+				return <a href="#" onClick={ this.onClickEnterPasswordInstead }>{ translate( 'Enter a password instead' ) }</a>;
+			}
+			if ( magicLoginView ) {
+				return;
+			}
+			footerLinks.push( <a href="#" onClick={ this.onMagicLoginRequestClick }>{ translate( 'Email me a login link' ) }</a> );
 		}
 
-		return (
-			<div className="wp-login__footer">
-				{ loginLink }
-				<a href={ config( 'login_url' ) + '?action=lostpassword' }>{ this.props.translate( 'Lost your password?' ) }</a>
-				<a href="#" onClick={ this.goBack }><Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Back' ) }</a>
-			</div>
+		footerLinks.push(
+			<a href={ config( 'login_url' ) + '?action=lostpassword' }>{ this.props.translate( 'Lost your password?' ) }</a>,
+			<a href="#" onClick={ this.goBack }><Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Back' ) }</a>
 		);
+
+		return footerLinks;
```

Comment:
I find this function harder to read because of the logic around the nested `if`s and the successive mutations of `footerLinks`. Consider:

```js
const requestPasswordButton = magicLoginEnabled && magicLoginView === REQUEST_FORM &&
  <a href="" …/>;
const requestLinkButton = magicLoginEnabled && ! magicLoginView
  <a href="" … />;

return compact( [
  requestPasswordButton,
  requestLinkButton,
  <a href="…" />,
  <a href="…" />,
] );
```